test/test_checks.py

```diff
@@ -1,13 +1,24 @@
-from flake8_quotes import QuoteChecker
 import os
+import re
 import subprocess
 from unittest import TestCase
 
+OUTPUT_REGEX = re.compile(r'.*?:([\d]+):([\d]+): (.*)')
```

Comment:
I'd really prefer to use `flake8` APIs instead of running the command directly =/ It slows down the tests and makes debugging weirder/harder (e.g. print statements are now in `assert` blocks instead of just `stdout`

Unfortunately, I took a shot but don't have time to dig to find a nice default options for `FileChecker` or `FileProcessor` =/

https://flake8.pycqa.org/en/3.8.2/internal/checker.html#processing-files

Reply:
Also, it's good practice to avoid using `.*` when you can. Better to use not class identifiers to avoid accidentally greedy code:

```suggestion
 OUTPUT_REGEX = re.compile(r'[^:]+:([\d]+):([\d]+): (.*)') 
```

Also can we add an example comment line before so someone can better understand what the regex is doing?

Reply:
The `.*?` is a non-greedy match, that change doesn't hurt but is unnecessary as the variants are functionally equivalent. If it's a preferred style thing, have at it, it's your project.

Personally I don't see the value in spending time optimizing unit test code, it's not something users will be dealing with and effort spent there should be balanced against developer pain and the amount of development effort going in to the project.